stats/record.go

```diff
@@ -22,10 +22,24 @@ import (
 	"go.opencensus.io/tag"
 )
 
+func init() {
+	internal.SubscriptionReporter = func(measure string) {
+		mu.Lock()
+		measures[measure].subscribe()
+		mu.Unlock()
+	}
+}
+
 // Record records one or multiple measurements with the same tags at once.
 // If there are any tags in the context, measurements will be tagged with them.
 func Record(ctx context.Context, ms ...Measurement) {
-	if len(ms) == 0 {
+	var filtered []Measurement
```

Comment:
I don't think it is worth it to construct a filtered slice here. I think we can make this even cheaper by:
1. Instead of having a subscribed call per measure, have `M` construct a `Measure` with nil `m` and define this to mean that the Measurement is ignored
2. Just do a pass over the `ms` and check if all are ignored and drop if that's the case, otherwise pass the original `ms` 
3. In the view code, filter the Measurements by `Measure() != nil` when processing view updates.

This would avoid allocations in `append` and also virtual calls to `subscribed`.

Reply:
Done